core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -88,6 +88,20 @@ class LogTest extends JUnitSuite {
       assertEquals("Changing time beyond rollMs and appending should create a new segment.", numSegments, log.numberOfSegments)
     }
 
+    time.sleep(log.config.segmentMs + 1)
+    val setWithTimestamp =
+      TestUtils.singleMessageSet(payload = "test".getBytes, timestamp = time.milliseconds + log.config.segmentMs + 1)
+    log.append(setWithTimestamp)
+    assertEquals("The number of segments should stay the same", 5, log.numberOfSegments)
```

Comment:
Ah... Sorry for the confusion, I changed the assertion expected value but forgot to change the assertion message.